shared/modules/Numeric.ts

```diff
@@ -0,0 +1,621 @@
+import { BigNumber } from 'bignumber.js';
+import BN from 'bn.js';
+import { isHexString, isNullOrUndefined } from '@metamask/utils';
+import { addHexPrefix } from 'ethereumjs-util';
+import { EtherDenomination } from '../constants/common';
+import { stripHexPrefix } from './hexstring-utils';
+
+export type NumericValue = string | number | BN | BigNumber;
+export type NumericBase = 10 | 16;
+
+/**
+ * All variations of isHexString from our own utilities and etherumjs-utils
+ * return false for a '-' prefixed hex string. This utility method strips the
+ * possible '-' from the string before testing its validity so that negative
+ * hex values can be properly handled.
+ *
+ * @param value - The string to check
+ * @returns true if the value is a hex string (negative or otherwise)
+ */
+function isHexStringOrNegatedHexString(value: string): value is string {
+  return isHexString(value.replace('-', '')) || isHexString(value);
+}
+
+/**
+ * BigNumber supports hex strings with '.' (aka decimals) in the string.
+ * No version of isHexString returs true if the string contains a decimal so
+ * this method is used to check if both parts of the string split by the
+ * decimal are hex strings. If so we can feed this value into BigNumber to get
+ * a valid Numeric.
+ *
+ * @param value - The string to check
+ * @returns true if the string is a hexadecimal split by '.'
+ */
+function isDecimalHex(value: string): boolean {
+  const parts = value.split('.');
+  if (parts.length === 1) {
+    return false;
+  }
+  return parts.every((part) => isHexStringOrNegatedHexString(part));
```

Comment:
this implementation will pattern match `-<hex num>.-<hex num>` as a decimal hex. Flagging to ensure it's intentional.

Reply:
It is a safety net. I don't think its actually valid hex to do `hex`.`hex` but BigNumber uses that notation internally for hexadecimal containing a decimal point when stringified.... it seems... There should be no other cases where we get 'hex'.'hex' other than as a result of toString on a BigNumber which will always be re-interpretable by BigNumber during instantiation. We should add tests for this soon but as of right now I am of the mind that this is okay.